snuba/datasets/configuration/storage_builder.py

```diff
@@ -92,7 +93,12 @@ def __build_readable_storage_kwargs(config: dict[str, Any]) -> dict[str, Any]:
         ),
         ALLOCATION_POLICY: AllocationPolicy.get_from_name(
             config[ALLOCATION_POLICY]["name"]
-        ).from_kwargs(**config[ALLOCATION_POLICY].get("args", {}))
+        ).from_kwargs(
+            **{
+                **config[ALLOCATION_POLICY].get("args", {}),
+                "storage_key": storage_key.value,
```

Comment:
Why does this take the `.value` instead of just the storage key?

Reply:
The `from_kwargs()` method expects string kwargs, and the method picks up the string and converts it back to a storage key object. Just seems like the general pattern for Registered classes, @volokluev can probably chime in here?